### PR TITLE
Sort out __proto__ occurrences outside of annexB

### PR DESCRIPTION
**Pull request by devsnek**
cc @shvaikalesh

### CODE REVIEW

**Comment by shvaikalesh** [comment]
I see some of changed tests check that `__proto__` is treated a regular property, while others rely on `Object.prototype.__proto__` getting/setting the `[[Prototype]]`. Maybe we should move to Annex B dir only the latter ones (Proxy tests)?

**Comment by devsnek** [comment]
the `__proto__` checks for json/regexp are testing that `__proto__` doesn't do something weird, which could only happen in an annex b environment. in a non-annex-b environment they're as meaningful as testing any other property name (not very meaningful)

File: test/built-ins/Proxy/set/call-parameters-prototype-dunder-proto.js [hunk 1]
```diff
@@ -1,56 +0,0 @@
-// Copyright (C) 2019 Alexey Shvayka. All rights reserved.
```
Comment (shvaikalesh) [inline test/built-ins/Proxy/set/call-parameters-prototype-dunder-proto.js:1]: This test file seems to be removed instead of moved to Annex B dir.

**Comment by rwaldron** [comment]
Please fix whatever the linter is failing on, then I will merge

**Pull request merged**